04-onion-routing.md

```diff
@@ -71,50 +71,74 @@ There are a number of conventions adhered to throughout this document:
  - The maximum route length is limited to 20 hops.
  - HMAC: the integrity verification of the packet is based on Keyed-Hash
  Message Authentication Code, as defined by the
- [FIPS 198 Standard](http://csrc.nist.gov/publications/fips/fips198-1/FIPS-198-1_final.pdf)/[RFC 2104](https://tools.ietf.org/html/rfc2104)
+ [FIPS 198 Standard](http://csrc.nist.gov/publications/fips/fips198-1/FIPS-198-1_final.pdf)/[RFC 2104](https://tools.ietf.org/html/rfc2104),
  and using a `SHA256` hashing algorithm.
  - Elliptic curve: for all computations involving elliptic curves, the
    Bitcoin curve is used, as specified in [`secp256k1`](http://www.secg.org/sec2-v2.pdf).
  - Pseudo-random stream: [`ChaCha20`](https://tools.ietf.org/html/rfc7539) is
  used to generate a pseudo-random byte stream. For its generation, a fixed
  null-nonce (`0x0000000000000000`) is used, along with a key derived from a
- shared secret and a `0x00`-byte stream of the desired output size as the message.
- - The terms _hop_ and _node_ are used interchangeably.
- - The term _peer_ refers only to a direct neighbor (in the overlay network) of the processing node.
+ shared secret and with a `0x00`-byte stream of the desired output size as the
+ message.
+
+ - The terms _origin node_ and _final node_ refer to the initial packet sender
+ and the final packet recipient, respectively.
+ - The terms _hop_ and _node_ are sometimes used interchangeably, but a _hop_
+ usually refers to an intermediate node in the route rather than an end node.
+        _origin node_ --> _hop_ --> ... --> _hop_ --> _final node_
+ - The term _processing node_ refers to the specific node along the route that is
+ currently processing the forwarded packet.
+ - The term _peers_ refers only to hops that are direct neighbors (in the
+ overlay network): more specifically, _sending peers_ forward packets to
+ _receiving peers_.
+        _sending peer_ --> _receiving peer_
```

Comment:
Should any of these additional terms be in the BOLT-1 glossary?

Reply:
I'd suggest origin node, processing node, and final node at a minimum. We should also clarify the "peer" definition with the sending/receiving dichotomy.

Reply:
sure I can add them to BOLT 1 if @rustyrussell agrees?

Reply:
Yes please.  As a separate PR, thanks!